lib/rubocop/cop/primer/primer_octicon.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+require "rubocop"
+require "primer/classify/utilities"
+
+# :nocov:
+module RuboCop
+  module Cop
+    module Primer
+      # This cop ensures that components use System Arguments instead of CSS classes.
+      #
+      # bad
+      # octicon("icon")
+      #
+      # good
+      # primer_octicon("icon")
```

Comment:
In the dotcom erblint I was transforming the icon to a symbol `octicon("icon") => primer_octicon(:icon)` thought that was preferable from a performance standpoint

Reply:
I decided to keep the icon as-is because I'm actually supporting ruby code here (ivar, method calls, ternaries) and it was kinda hard to figure out if the icon was a plain string or something else, I'll give it another try

Reply:
I updated it to deal with strings when passed directly, but kept ternaries untouched